src/app/utils/getRPCAccountBalances.ts

```diff
@@ -0,0 +1,47 @@
+import { Runtime, RuntimeSdkBalance } from '../../oasis-nexus/api'
+import * as oasis from '@oasisprotocol/client'
+import * as oasisRT from '@oasisprotocol/client-rt'
+import { Network } from '../../types/network'
+import { getTokensForScope, paraTimesConfig } from '../../config'
+import { fromBaseUnits } from './helpers'
+
+const grpcUrls: { [key in Network]: string } = {
+  [Network.mainnet]: 'https://grpc.oasis.dev',
+  [Network.testnet]: 'https://testnet.grpc.oasis.dev',
+}
+
+export async function getRPCAccountBalances(
+  oasisAddress: string,
+  scope: {
+    network: Network
+    layer: Runtime
+  },
+): Promise<RuntimeSdkBalance[]> {
+  const paratimeConfig = paraTimesConfig[scope.layer][scope.network]
+  if (!paratimeConfig.runtimeId) throw new Error('Paratime is not configured')
+
+  const nic = new oasis.client.NodeInternal(grpcUrls[scope.network])
+  const accountsWrapper = new oasisRT.accounts.Wrapper(oasis.misc.fromHex(paratimeConfig.runtimeId))
+
+  const rawBalances = (
+    await accountsWrapper
+      .queryBalances()
+      .setArgs({ address: oasis.staking.addressFromBech32(oasisAddress) })
+      .query(nic)
+  ).balances
+
+  return [...rawBalances.entries()]
+    .map(([denomination, amount]) => {
+      return {
+        balance: oasis.quantity.toBigInt(amount).toString(),
+        token_symbol: oasis.misc.toStringUTF8(denomination) || getTokensForScope(scope)[0].ticker,
+        token_decimals: paraTimesConfig[scope.layer].decimals, // TODO: differentiate by token
```

Comment:
That doesn't look right, if this is querying ERC-20 tokens as well, each of the ERC-20 token, can have custom decimals specified. So defaulting to paratime decimals here, is wrong.

Reply:
It's only querying e.g. TEST and EUROe; evm_balances are a different field in Nexus API.
But we don't differentiate decimals between TEST and EUROe - related: https://github.com/oasisprotocol/explorer/pull/1265#discussion_r1498078526

Reply:
Ok, I verified that both Ocean and EUROe use 18 decimals places, so coincidentally this works as expected.

Looking at the documentation at https://docs.pontus-x.eu/docs/Development/quick_start#setup-metamask the native token has been renamed to GX @csillag are we aware of this?